frontend/javascripts/oxalis/api/api_latest.js

```diff
@@ -693,9 +696,9 @@ class DataApi {
    */
   getVolumeTracingLayerName(): string {
     // TODO: Rename method to getSegmentationLayerName() and increase api version
-    const segmentationLayer = this.model.getSegmentationLayer();
-    assertExists(segmentationLayer, "Segmentation layer not found!");
-    return segmentationLayer.name;
+    const segmentationLayerName = this.model.getSegmentationLayerName();
+    assertExists(segmentationLayerName, "Segmentation layer not found!");
+    return segmentationLayerName || "";
```

Comment:
I need the alternative of  `""` here as flow does not know that at this point `segmentationLayerName` is not null.

Is this fine how I handled this or should I revert this change and use the old version.

This change here is just some refactoring that came into my mind.

Reply:
I'd replace `assertExists` with a normal if-null-then-throw logic. That way, flow will understand that the following code can safely access `segmentationLayerName`.

```
if (!segmentationLayerName) {
  throw new Exception("Segmentation layer not found!");
}
return segmentationLayerName;
```